src/test/ui/parser/issue-81097.stderr

```diff
@@ -0,0 +1,8 @@
+error: expected one of `;` or `]`, found `,`
+  --> $DIR/issue-81097.rs:2:15
+   |
+LL |     drop::<[(), 0]>([]);
+   |               ^ expected one of `;` or `]`
```

Comment:
In the nightly release we currently also get the `^ expected one of ';' or ']'` error, the first error we currently get is misleading though. But I would argue that a little thought easily solves this initial confusion in all cases. I don't want to sound too negative here and the rust community appreciates your effort, but I feel that the overhead of reading your additional code in the parser is not worth the benefit of having the misleading first error message removed.

Reply:
> I feel that the overhead of reading your additional code in the parser is not worth the benefit of having the misleading first error message removed.

I'd be happy to implement this differently if anyone has any suggestions. I'm quite new here and I don't know the code in detail.

Regarding the extra code, I'd leave that decision to the active maintainers/developers of the parser.

In compiler development this kind of tradeoff exists all the time: you add code for user convenience but it makes maintainers' job more difficult. I don't know how rustc devs decides on what to do in these cases but I'm happy to close the PR if they think this code is not worth it. (I guess in that case the issue should be closed too?)

I made the mistake of using `,` instead of `;` in array types in the past many times, but for me the second message is clear enough and the first one doesn't bother me too much. So I don't feel strongly either way.